app/src/main/java/com/example/trainingintive/dog_images_feature/data/repository/DogImageRepository.kt

```diff
@@ -12,8 +14,12 @@ class DogImageRepository @Inject constructor(
 ) {
 
     fun getDogImageUrl() =
-        dogImageApiService.getDogImage()
-            .map { it.toDomain() }
+        getLastPosition().flatMap {
```

Comment:
Can be simplified:
```
 getLastPosition()
    .flatMap { position ->
        dogImageApiService.getDogImage()
            .map { it.toDomain(position)}
    }
```
But it will be best to extract `getDogImage` part to some private function for simplicity, like this

```
 getLastPosition()
    .flatMap { getDogImage(it) }
```
where `getDogImage` is your private fun